yt_dlp/extractor/rcti.py

```diff
@@ -0,0 +1,242 @@
+        'D': 18,
+    }
+
+    def _entries(self, url, display_id=None, note='Downloading entries JSON', metadata={}):
+        total_pages = 0
+        try:
+            total_pages = self._call_api(
+                '%s&length=20&page=0' % url,
+                display_id, note)[1]['pagination']['total_page']
+        except ExtractorError as e:
+            if 'not found' in str(e):
+                return []
+            raise e
+        if total_pages <= 0:
+            return []
+
+        for page_num in range(1, total_pages + 1):
+            episode_list = self._call_api(
+                '%s&length=20&page=%s' % (url, page_num),
+                display_id, '%s page %s' % (note, page_num))[0] or []
+
+            for video_json in episode_list:
+                link = video_json['share_link']
+                url_res = self.url_result(link, 'RCTIPlus', video_json.get('product_id'), video_json.get('title'))
+                url_res.update(metadata)
+                yield url_res
+
+    def _real_extract(self, url):
+        series_id, display_id = re.match(self._VALID_URL, url).groups()
+
+        series_meta, meta_paths = self._call_api(
+            'https://api.rctiplus.com/api/v1/program/%s/detail' % series_id, display_id, 'Downloading series metadata')
+        metadata = {
+            'age_limit': try_get(series_meta, lambda x: self._AGE_RATINGS[x['age_restriction'][0]['code']])
+        }
+
+        cast = []
+        for star in series_meta.get('starring', []):
+            cast.append(strip_or_none(star.get('name')))
+        for star in series_meta.get('creator', []):
+            cast.append(strip_or_none(star.get('name')))
+        for star in series_meta.get('writer', []):
+            cast.append(strip_or_none(star.get('name')))
+        metadata['cast'] = cast
+
+        tags = []
+        for tag in series_meta.get('tag', []):
+            tags.append(strip_or_none(tag.get('name')))
+        metadata['tag'] = tags
+
+        entries = ()
+        seasons_list = self._call_api(
+            'https://api.rctiplus.com/api/v1/program/%s/season' % series_id, display_id, 'Downloading seasons list JSON')[0]
+        for season in seasons_list:
+            entries += (self._entries('https://api.rctiplus.com/api/v2/program/%s/episode?season=%s' % (series_id, season['season']),
+                                      display_id, 'Downloading season %s episode entries' % season['season'], metadata),)
+
+        entries += (self._entries('https://api.rctiplus.com/api/v2/program/%s/clip?content_id=0' % series_id,
+                                  display_id, 'Downloading clip entries', metadata),)
+        entries += (self._entries('https://api.rctiplus.com/api/v2/program/%s/extra?content_id=0' % series_id,
+                                  display_id, 'Downloading extra entries', metadata),)
+
+        return self.playlist_result(itertools.chain(*entries), series_id, series_meta.get('title'), series_meta.get('summary'))
```

Comment:
```suggestion
        return self.playlist_result(itertools.chain(*entries), series_id, series_meta.get('title'), series_meta.get('summary'), **metadata)
```

since the metadata applies for all videos in playlist, it would be nice to have in the playlist infojson as well